### ISSUE TITLE
Get-BCArtifactUrl has become a lot slower with 2.0.0

### ISSUE THREAD

**Issue by tfenster** [microsoft/navcontainerhelper#1646]
**Describe the issue**
After updating bccontainerhelper to 2.0.0, Get-BCArtifactUrl seems to have become a lot slower. The following tests are done in containers on the same host, so CPU / RAM / disk / network are identical

**Scripts used to create container and cause the issue**
Old version, 12 seconds for the first call, 9 for the second
```
PS C:\> import-module bccontainerhelper
BcContainerHelper version 1.0.16
WARNING: The names of some imported commands from the module 'bccontainerhelper' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run 
the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
PS C:\> Measure-Command {Get-BCArtifactUrl -country de -storageAccount bcartifacts -type sandbox}
Days              : 0
Hours             : 0
Seconds           : 12
Milliseconds      : 166
Ticks             : 121667027
TotalDays         : 0.000140818318287037
TotalHours        : 0.00337963963888889
TotalMinutes      : 0.202778378333333
TotalSeconds      : 12.1667027
TotalMilliseconds : 12166.7027

PS C:\> Measure-Command {Get-BCArtifactUrl -country de -storageAccount bcartifacts -type sandbox}
Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 9
Milliseconds      : 170
Ticks             : 91700488
TotalDays         : 0.000106134824074074
TotalHours        : 0.00254723577777778
TotalMinutes      : 0.152834146666667
TotalSeconds      : 9.1700488
TotalMilliseconds : 9170.0488
```

New version, first call 99 seconds, second call 136 seconds
```
PS C:\> import-module bccontainerhelper
BcContainerHelper version 2.0.0
WARNING: The names of some imported commands from the module 'bccontainerhelper' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run 
the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
PS C:\> Measure-Command {Get-BCArtifactUrl -country de -storageAccount bcartifacts -type sandbox}
Days              : 0
Hours             : 0
Minutes           : 1
Seconds           : 39
Milliseconds      : 334
Ticks             : 993343631
TotalDays         : 0.00114970327662037
TotalHours        : 0.0275928786388889 
TotalMinutes      : 1.65557271833333   
TotalSeconds      : 99.3343631
TotalMilliseconds : 99334.3631

PS C:\> Measure-Command {Get-BCArtifactUrl -country de -storageAccount bcartifacts -type sandbox}
Days              : 0
Hours             : 0
Minutes           : 2
Seconds           : 16
Milliseconds      : 134
Ticks             : 1361340404
TotalDays         : 0.00157562546759259
TotalHours        : 0.0378150112222222
TotalMinutes      : 2.26890067333333
TotalSeconds      : 136.1340404
TotalMilliseconds : 136134.0404
```

**Full output of scripts**
see above

**Screenshots**
-

**Additional context**
- does it happen all the time?
so far, yes

**Comment by freddydk** [comment]
What OS are you running?

**Comment by freddydk** [comment]
Ok, I can repro this - unsure why I haven't seen his before:-(

**Comment by freddydk** [comment]
This was caused by adding the before and after:-(
This code:
```
$enumerationResults.Blobs.Blob | % {
    $Artifacts += $_.Name
}
```
is THAT much slower than this code:
```
$Artifacts += $enumerationResults.Blobs.Blob.Name
```
I learned something new today.

**Comment by freddydk** [comment]
Fixed in 2.0.1

**Comment by tfenster** [comment]
Confirmed, thanks!

```
PS C:\> Import-Module BcContainerHelper 
BcContainerHelper version 2.0.1
WARNING: The names of some imported commands from the module 'BcContainerHelper' include unapproved verbs that might make them    
less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For
 a list of approved verbs, type Get-Verb.
PS C:\> Measure-Command {Get-BCArtifactUrl -country de -storageAccount bcartifacts -type sandbox}
Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 11
Milliseconds      : 671
Ticks             : 116715625
TotalDays         : 0.000135087528935185
TotalHours        : 0.00324210069444444 
TotalMinutes      : 0.194526041666667   
TotalSeconds      : 11.6715625
TotalMilliseconds : 11671.5625

PS C:\> Measure-Command {Get-BCArtifactUrl -country de -storageAccount bcartifacts -type sandbox}
Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 14
Milliseconds      : 450
Ticks             : 144502713
TotalDays         : 0.000167248510416667
TotalHours        : 0.00401396425
TotalMinutes      : 0.240837855
TotalSeconds      : 14.4502713
TotalMilliseconds : 14450.2713
```
